app/models/user.rb

```diff
@@ -0,0 +1,28 @@
+class User < ActiveRecord::Base
+  has_and_belongs_to_many :twitter_users
+  has_and_belongs_to_many :vimeo_users
+
+  validates :name, presence: true
+  validates :uid, presence:true, uniqueness: true
+  validates :provider, presence: true
+
+  def self.find_or_create_from_omniauth(auth_hash)
+    user = self.find_by(uid: auth_hash["uid"], provider: auth_hash["provider"])
+    if !user.nil?
```

Comment:
you can just use `if user` which would evaluation to true and could return the user, I would likely replace most of this method with `self.where(uid: auth_hash["uid"], provider: auth_hash["provider"].first_or_create(attributes)` per http://apidock.com/rails/ActiveRecord/Relation/first_or_create, which would find the user if it exists, or create it if it doesn't